x/ibc/light-clients/07-tendermint/types/upgrade.go

```diff
@@ -20,14 +23,24 @@ import (
 //   and ProofSpecs do not match parameters set by committed client
 func (cs ClientState) VerifyUpgrade(
 	ctx sdk.Context, cdc codec.BinaryMarshaler, clientStore sdk.KVStore,
-	upgradedClient exported.ClientState, proofUpgrade []byte,
+	upgradedClient exported.ClientState, upgradeHeight exported.Height, proofUpgrade []byte,
 ) error {
-	if cs.UpgradePath == nil {
+	if cs.UpgradePath == "" {
 		return sdkerrors.Wrap(clienttypes.ErrInvalidUpgradeClient, "cannot upgrade client, no upgrade path set")
 	}
+	// construct MerklePath from upgradePath
+	upgradeKeys := strings.Split(cs.UpgradePath, "/")
+	// append upgradeHeight to last key in merkle path
+	upgradeKeys[len(upgradeKeys)-1] = fmt.Sprintf("%s/%d", upgradeKeys[len(upgradeKeys)-1], upgradeHeight.GetEpochHeight())
+	upgradePath := commitmenttypes.NewMerklePath(upgradeKeys)
+
+	if cs.GetLatestHeight().GetEpochNumber() != upgradeHeight.GetEpochNumber() {
```

Comment:
could you add a comment that the version number is verified implicitly by using the consensus state root that belongs to that height. It wasn't immediately clear to me why you don't need the version number in the upgrade height setting the upgrade client